src/engine/order.ts

```diff
@@ -47,53 +41,58 @@ export class OrderEngine {
 
   public async executeBuy(quote?: number, strength = 1) {
     const { exchange, symbol } = this.opts
-    const price = this.exchangeProvider.priceToPrecision(exchange, symbol, quote ? quote : await this.quoteEngine.getBuyPrice())
+    const rawPrice = quote ? quote : await this.quoteEngine.getBuyPrice()
+    const price = this.exchangeProvider.priceToPrecision(exchange, symbol, rawPrice)
```

Comment:
It should be possible to change the return type of this Method in the ExchangeWrapper to number, shouldn't it? Afterwards typescript could correctly determinate the type.
Afterwards the number-conversion in line 138 is not needed any longer, or am i missing something?